### DISCUSSION TITLE
strongswan networkmanager plugin bypass identity constraint (rightid)

### DISCUSSION THREAD

**Discussion by wizche** [strongswan/strongswan#1333] [Q&A]
Hello,

I have a very simple question, when using the network manager plugin, how can I get the `rightid=%any` propagated? (to bypass the `constraint check failed: identity 'mydomain.com' required`)

```
[vpn]
address=example.com
encap=no
ipcomp=no
method=psk
password-flags=1
proposal=no
remote-identity=%any 
virtual=no
service-type=org.freedesktop.NetworkManager.strongswan
``` 
I tried settings the `remote-identity` to `%any` but I still get the constraint error.

Whereas with a corresponding connection config like this will works:
```
conn myvpn
        keyexchange=ikev2
	leftsourceip=%config
	right=mydomain.com
	rightid=%any
	rightauth=psk
	auto=add
```

**Comment by tobiasbrunner** [comment]
`%any` is explicitly ignored by the NM plugin to avoid users from authenticating against any random server certificate (the plugin defaults to trust all system-installed CA certificates, so the client e.g. trusts any random Let's Encrypt cert, that check at least ensures the domain matches).

Why don't you fix your certificate by including `mydomain.com` as subjectAltName extension?

**Reply by wizche** [reply]
It was more meant as temporary solution... I was wondering if there was a straight up way to just bypass that temporarily...
